bigquery_etl/query_scheduling/task.py

```diff
@@ -103,6 +103,7 @@ class Task:
     multipart: bool = attr.ib(False)
     sql_file_path: Optional[str] = None
     priority: Optional[int] = None
+    referenced_tables: Optional[List[Tuple[str, str]]] = attr.ib(None)
```

Comment:
Why is this a list of 2-tuples rather than a map?

Reply:
it's the return type of `self._get_referenced_tables(client)`, which is a list of table references as a tuple of `dataset_id` and `table_id`. neither `dataset_id` nor `table_id` would be a unique key for each element.